Dynamic code snippet for interactive documentation

StephaneBranly opened the pull request:
Hello

I've been using dash mantine for almost 2 years now, and I really appreciate the interactive part of the documentation that we're used to finding on component libraries.

I'd like to suggest this feature, which provides a code snippet of what the user has configured.

I'm available for any updates, as there may be patches to apply.

![image](https://github.com/user-attachments/assets/9650976f-3ee1-4113-9cec-67728bd85648)

AnnMarieW commented:
Wow!  This is awesome :star_struck:   
This is a nice addition - it's nice to see the actual code for the interactive examples.

I'm just starting to dig in so I'll have continue to post more comments as I notice things.
- Some examples such as Button, Chip, Burger, ThemeIcon (and probably others?) use  `dmc.Center` instead of the correct component
- The `Stepper` example doesn't render correctly until after I change one of the selections.  Might be because there are two interactive examples on the same page.  The second example renders correctly.
- Need to remove the console.log before merging.

StephaneBranly commented:
> Wow! This is awesome 🤩 This is a nice addition - it's so helpful to see the actual code for the interactive examples.
> 
> I'm just starting to dig in so I'll continue to add to this list as I notice things.
> 
>     * Some examples such as Button, Chip, Burger, ThemeIcon (and probably others?) use  `dmc.Center` instead of the correct component.  Looks like it's because the target component in the interactive exaple  wraps the example component in a `dmc.Center`
> 
>     * The `Stepper` example doesn't render correctly until after I change one of the selections.  Might be because there are two interactive examples on the same page.  The second example renders correctly.
> 
>     * Need to remove the console.log before merging.
> 
>     * Since the code snippet shows only the props that are configured in the interactive example, it might be a little confusing for components like Accordion or the graph components that require other props to make the example run.  Can you think of a clear and concise way to make that clear?

Thanks for the feedback !

* I'm working on a refactor and fix concerning some components wrapped in a parent class
* I found the source error, https://github.com/snehilvj/dmc-docs/blob/aff1c694e5ab6579c26477d760251bcbca94e830/lib/configurator.py#L65 ; corrected by setting with `value` instead of the `mapped[value]`
* Of course, removed
* I will check Mantine v7 doc to see how the deal with

I've noticed additionnel points :
* I can add code tabs with python logo like Mantine v7 doc
* Sometimes, this dynamic doc generates a duplicate of a static example

StephaneBranly commented:
I've just added these comments
![image](https://github.com/user-attachments/assets/084ce792-4888-4310-a8ca-1a8d189dead3)

AnnMarieW commented:
This looks fantastic!
The inline comments in the code snippets looks good. I’ll test a few more pages, but everything’s looking solid—planning to merge it this afternoon.

Thanks so much for the great work on this feature. :woman_dancing:

AnnMarieW commented:
Nice catch on the Stepper.

Looks like the Tooltip page has some duplicate static examples

StephaneBranly commented:
> Nice catch on the Stepper.
> 
> Looks like the Tooltip page has some duplicate static examples

Good point, I've just corrected it
